src/EditorFeatures/Core/Implementation/CommentSelection/AbstractCommentSelectionBase.cs

```diff
@@ -0,0 +1,212 @@
+                    {
+                        Format(service, subjectBuffer.CurrentSnapshot, trackingSnapshotSpans, CancellationToken.None);
+                        transaction.Complete();
+                    }
+                }
+
+                // Set the selection after the edits have been applied.
+                var spansToSelect = trackingSnapshotSpans.Select(s => new Selection(s));
+                textView.GetMultiSelectionBroker().SetSelectionRange(spansToSelect, spansToSelect.Last());
+            }
+        }
+
+        /// <summary>
+        /// Creates a tracking span for the operation.
+        /// Internal for tests.
+        /// </summary>
+        internal static ITrackingSpan CreateTrackingSpan(Operation operation, ITextSnapshot snapshot, TextSpan textSpan)
+        {
+            var spanTrackingMode = operation == Operation.Comment
+                ? SpanTrackingMode.EdgeInclusive
+                : SpanTrackingMode.EdgeExclusive;
+            return snapshot.CreateTrackingSpan(Span.FromBounds(textSpan.Start, textSpan.End), spanTrackingMode);
+        }
+
+        /// <summary>
+        /// Retrieves the snapshot span from a post edited tracking span.
+        /// Additionally applies any extra modifications to the tracking span post edit.
+        /// Internal for tests.
+        /// </summary>
+        private static SnapshotSpan CreateSnapshotSpan(ITextSnapshot snapshot, ITrackingSpan trackingSpan, CommentTrackingSpan originalSpan)
+        {
+            var snapshotSpan = trackingSpan.GetSpan(snapshot);
+            if (originalSpan.HasPostApplyChanges())
+            {
+                var updatedStart = snapshotSpan.Start.Position + originalSpan.AmountToAddToStart;
+                var updatedEnd = snapshotSpan.End.Position + originalSpan.AmountToAddToEnd;
+                if (updatedStart >= snapshotSpan.Start.Position && updatedEnd <= snapshotSpan.End.Position)
+                {
+                    snapshotSpan = new SnapshotSpan(snapshot, Span.FromBounds(updatedStart, updatedEnd));
+                }
+            }
+
+            return snapshotSpan;
+        }
+
+        private static void Format(ICommentSelectionService service, ITextSnapshot snapshot, IEnumerable<SnapshotSpan> changes, CancellationToken cancellationToken)
+        {
+            var document = snapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document == null)
+            {
+                return;
+            }
+
+            var textSpans = changes.Select(change => change.Span.ToTextSpan()).ToImmutableArray();
+            var newDocument = service.FormatAsync(document, textSpans, cancellationToken).WaitAndGetResult(cancellationToken);
+            newDocument.Project.Solution.Workspace.ApplyDocumentChanges(newDocument, cancellationToken);
+        }
+
+        private static ICommentSelectionService GetService(Document document)
```

Comment:
this methods seems entirely inlinable as just `document.GetLanguageService<ICommentSelectionService>()`...

Reply:
Yep, artifact of removing the old service.  Will inline.